### ISSUE TITLE
mapfishapp: shp/kml file upload w/ gdal.jar broken ? (update geotools ?)

### ISSUE THREAD

**Issue by landryb** [georchestra/georchestra#2121]
cf #2046 & af15580, and it seems there are numerous issues with mfapp upload (a dozen or so if you look for 'upload' in opened issues), it seems right now file upload doesnt work if gdal.jar is detected by mfapp. Uploading a gpx or a zipped shp only triggers the following error message on stderr (ie catalina.out):

```
ERROR 1: SQL Expression Parsing Error: syntax error, unexpected end of string, expecting '.'. Occurred around :
SELECT FID, * FROM 'ETABLISSEMENT' 
```
the error message comes from libgdal.so, same thing with 2.2.3 (local) or 2.1.2 (default version in debian stretch) - maybe related to the geotools 9.2 shipped with mfapp...

gml upload works.

Removing gdal.jar from the context (ie i had copied it from usr/share/java/gdal.jar coming from libgdal-java) to mapfishapp/WEB-INF/lib) fixes shp upload as it fallbacks to pure java/geotools implem, but then you lose the ability to upload osm/gpx/etc..

**Comment by landryb** [comment]
it *seems* geotools expects gdal 'around' 1.8.1, per https://github.com/geotools/geotools/blob/master/modules/unsupported/ogr/ogr-jni/pom.xml#L84 - and the ogr plugin is back in the unsupported category  upstream since https://osgeo-org.atlassian.net/browse/GEOT-6001 / https://github.com/geotools/geotools/pull/1872

**Comment by landryb** [comment]
upload of zipped mif/mid with the geotools implem fails : 
```
java.io.IOException: The EPSG code wasn't found for the mif coordinate system: 3, 33, "m", 3, 46.5, 44, 49, 700000, 6600000
        at org.geotools.data.mif.MIFProjReader.toCRS(MIFProjReader.java:249)
        at org.geotools.data.mif.MIFFile.retrieveSRID(MIFFile.java:333)
        at org.geotools.data.mif.MIFFile.<init>(MIFFile.java:286)
        at org.geotools.data.mif.MIFDataStore.registerMIF(MIFDataStore.java:264)
        at org.geotools.data.mif.MIFDataStore.<init>(MIFDataStore.java:107)
```
not sure what should be done for this one..

**Comment by landryb** [comment]
One of the drawbacks of not having gdal.jar enabled in mfapp, and the detection code is trying to load the class: each time someone hits mapfishapp homepage, a huge traceback is puked in its log:

```
2018-07-17 13:20:25 upload [INFO] gdal/ogr is not available in the system
java.lang.NoClassDefFoundError: Could not initialize class org.geotools.data.ogr.jni.JniOGR
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:264)
        at org.georchestra.mapfishapp.ws.upload.OGRFeatureReader.isOK(OGRFeatureReader.java:261)
        at org.georchestra.mapfishapp.ws.upload.AbstractFeatureGeoFileReader.createImplementationStrategy(AbstractFeatureGeoFileReader.java:202)   
        at org.georchestra.mapfishapp.ws.upload.AbstractFeatureGeoFileReader.<init>(AbstractFeatureGeoFileReader.java:75)
        at org.georchestra.mapfishapp.ws.AbstractController.defaultModel(AbstractController.java:73)
        at org.georchestra.mapfishapp.ws.AbstractController.createModelFromString(AbstractController.java:51)
        at org.georchestra.mapfishapp.ws.HomeController.handleGETRequest(HomeController.java:76)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.springframework.web.bind.annotation.support.HandlerMethodInvoker.invokeHandlerMethod(HandlerMethodInvoker.java:176)
        at org.springframework.web.servlet.mvc.annotation.AnnotationMethodHandlerAdapter.invokeHandlerMethod(AnnotationMethodHandlerAdapter.java:436)
        at org.springframework.web.servlet.mvc.annotation.AnnotationMethodHandlerAdapter.handle(AnnotationMethodHandlerAdapter.java:424)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:900)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:827)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:882)
        at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:778)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:635)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:742)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:728)
        at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:467)
        at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:392)
        at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:311)
        at org.tuckey.web.filters.urlrewrite.NormalRewrittenUrl.doRewrite(NormalRewrittenUrl.java:195)
        at org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:159)
        at org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:141)
        at org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:90)
        at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:406)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:478)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140)
        at psiprobe.Tomcat85AgentValve.invoke(Tomcat85AgentValve.java:35)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:80)
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:624)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
        at org.apache.catalina.ha.tcp.ReplicationValve.invoke(ReplicationValve.java:322)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:799)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:861)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1455)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:748)
2018-07-17 13:20:25 upload [INFO] It was set: org.georchestra.mapfishapp.ws.upload.GeotoolsFeatureReader
2018-07-17 13:20:25 upload [INFO] Using implementation: org.georchestra.mapfishapp.ws.upload.GeotoolsFeatureReader
```

Nicely handling this fallback case without the traceback would be great :)

**Comment by landryb** [comment]
Rereading the code and doing more testing, i cant reproduce the last error/traceback on master, only on my 'old' 15.12 installs (dev & prod), but the corresponding code under upload/ didnt change since a while... so maybe something changed in mfapp init, or this code isnt used anymore. I cant even trigger the `upload [INFO] Using implementation` logging..

**Comment by fvanderbiest** [comment]
@cmangeat can you have a look at the upgrade path for geotools in mapfishapp ?

**Comment by fvanderbiest** [comment]
Well, as discussed with @cmangeat today, upgrading geotools will not fix the issue since, as you said, the ogr module is now unsupported and requires older gdal versions.

Maybe time to drop the implementation relying on native libs ?

**Comment by bchartier** [comment]
Would it imply a loss of support for some geodata formats (for file upload function of mapfishapp)?

**Comment by landryb** [comment]
It doesnt work right now, so it cant be worse in terms of formats supported...

**Comment by bchartier** [comment]
ok, je me casse

**Comment by fvanderbiest** [comment]
No no @bchartier, you are a valuable contributor :-) 
You're right we will loose the ability to upload GPX & OSM files.
I'm a bit surprised though because IIRC a Geotools GPX module exists.

**Comment by fvanderbiest** [comment]
From 16.12 on, our newest mapfishapp docker images will not ship gdal anymore.
Still to be done: expunging related code from mapfishapp.

**Comment by fvanderbiest** [comment]
> Still to be done: expunging related code from mapfishapp.

Work in progress by @cmangeat

**Comment by cmangeat** [comment]
Seems that https://github.com/georchestra/georchestra/commit/69b67f8da776e30c3718e588d5c45b59f4e7a38b helps importing mif/mids (tested with attached files).

**Comment by fvanderbiest** [comment]
#2191 removes native libs support for mapfishapp on 16.12
Cascade merged to 17.12 with bdb010e841988dd83d589104cb56768bd877e60e
Merging to 18.06 was unsuccessful on first try (warning: inexact rename detection was skipped due to too many files.) ... will require a second look.

**Comment by fvanderbiest** [comment]
Now fixed on 18.06 and master
